packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -97,29 +97,89 @@ describe("Presence", () => {
 				// initialization, but should go unnoticed by the presence manager
 				// until there is a join signal related to it.
 				const rejoinAttendeeConnectionId = "client7";
-				let initialAttendeeSignal: ReturnType<typeof generateBasicClientJoin>;
-				let rejoinAttendeeSignal: ReturnType<typeof generateBasicClientJoin>;
+
+				let initialAttendeeSignalInfo: IAttendeeSignalInfo;
+				let rejoinAttendeeSignalInfo: IAttendeeSignalInfo;
+				interface IAttendeeSignalInfo {
+					clientSessionId: string;
+					clientConnectionId: ClientConnectionId;
+					fixedTime: number;
+					averageLatency?: number;
+					connectionOrder?: number;
+					updateProviders?: string[];
+					priorClientToSessionId?: Record<
+						ClientConnectionId,
+						{ rev: number; timestamp: number; value: string }
+					>;
+				}
+
+				function simulateAttendeeJoin(
+					attendeeSignalInfo: IAttendeeSignalInfo[],
+				): ISessionClient[] {
+					const joinedAttendees: ISessionClient[] = [];
+					const signals = attendeeSignalInfo.map((info) => {
+						return generateBasicClientJoin(info.fixedTime, {
+							...info,
+						});
+					});
+
+					const cleanUpListener = presence.events.on("attendeeJoined", (attendee) => {
+						joinedAttendees.push(attendee);
+					});
+
+					afterCleanUp.push(cleanUpListener);
+
+					for (const signal of signals) {
+						presence.processSignal("", signal, false);
+					}
+
+					return joinedAttendees;
+				}
+
+				function verifyAttendee(
+					attendee: ISessionClient,
+					connectionId: ClientConnectionId,
+					connectionStatus: SessionClientStatus = SessionClientStatus.Connected,
```

Comment:
Since one of these is the thing checked and the others are expectations, it would be helpful to name them that way.
`actualAttendee`
`expectedConnectionId`
`expectedConnectionStatus`